### PR TITLE
Split cron jobs and move libs to releases

### PR DESCRIPTION
**Pull request by lucasssvaz**
Split the cron job using a matrix with preparations for release/v5.3

### CODE REVIEW

**Comment by me-no-dev** [comment]
Have a look at [this file](https://github.com/espressif/esp32-arduino-lib-builder/blob/master/tools/check-deploy-needed.sh) which should run in the first job to determine if the cron should even run. Yaml code for it was commented in the cron workflow

**Comment by lucasssvaz** [comment]
Isnt this already being done here ? https://github.com/espressif/esp32-arduino-lib-builder/blob/master/tools/install-esp-idf.sh#L54-L97

**Comment by lucasssvaz** [comment]
@me-no-dev PTAL if the libs generated [here](https://github.com/lucasssvaz/esp32-arduino-lib-builder/actions/runs/9754805059) looks fine. Ill also open a PR with some changes to the IDF4 branch here so we can manually trigger it if needed.

Also, we might not be able to use environment variables in the manifest as it [is not allowed by the IDF registry](https://docs.espressif.com/projects/idf-component-manager/en/latest/reference/manifest_file.html#environment-variables-in-manifest). Do you have any other ideas on how to do what we discussed ?

**Comment by me-no-dev** [comment]
> Do you have any other ideas on how to do what we discussed ?

I suggest you ask in MM if there is any other way. That was the only thing I heard

File: .github/workflows/cron.yml [hunk 1]
```diff
@@ -1,144 +1,197 @@
...
+    name: Generate matrix
     runs-on: ubuntu-latest
-    
+    outputs:
+      matrix: ${{ steps.gen-matrix.outputs.matrix }}
+      branches: ${{ steps.gen-matrix.outputs.branches }}
+    steps:
+      - name: Generate matrix
+        id: gen-matrix
+        run: |
```
Comment (me-no-dev) [inline .github/workflows/cron.yml:44]: can this be a script?

Comment (lucasssvaz) [reply .github/workflows/cron.yml:44]: Yeah, didnt think about that. It will be even easier to maintain

**Comment by lucasssvaz** [comment]
@me-no-dev PTAL if the generated libs look correct. Specially the package JSON file.
https://github.com/lucasssvaz/esp32-arduino-lib-builder/actions/runs/9786603646

**Comment by me-no-dev** [comment]
json looks good

File: .github/workflows/cron.yml (hunk 1 above)
Comment (me-no-dev) [reply .github/workflows/cron.yml:39]: you marked as resolved, but this is still not a script, rather inline

**Comment by me-no-dev** [comment]
most of the things we discussed are yet to be implemented :)

**Comment by lucasssvaz** [comment]
@me-no-dev Moved to using releases. Tests in my forks appear to be working as expected

https://github.com/lucasssvaz/esp32-arduino-lib-builder/actions/runs/9966497424/job/27539203528
https://github.com/lucasssvaz/esp32-arduino-libs/releases/tag/idf-release_v5.1
https://github.com/lucasssvaz/arduino-esp32/pull/31

**Pull request merged**